### PR TITLE
Add SparkPi Vert.x examples

### PR DESCRIPTION
**Pull request by ruivieira**
Code examples for Java SparkPi with Vert.x

### CODE REVIEW

**Comment by ruivieira** [comment]
@elmiko ptal

**Comment by elmiko** [comment]
@ruivieira this looks great, thanks for the writeup. a couple things we need to fix

1. remove "WIP" from the commit title
2. change the spark version in the pom.xml file from `2.2.0` to `2.3.0`
3. chase down this bug in the code

when i run the application created by these files, i see this error as the app launches and attaches to the spark cluster. mind you, i have already replaced the spark version in my pom.xml before building this.

```
spark-submit --class io.vertx.core.Launcher --master spark://cluster-0d15f2:7077 --conf spark.driver.host=vertx-sparkpi-headless --driver-java[2/9560]
="-Dvertx.cacheDirBase=/tmp/vertx-cache" /opt/app-root/src/sparkpi-app-1.0-SNAPSHOT-vertx.jar                                                        
18/07/10 12:40:20 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable       
java.lang.NullPointerException                                                                                                                       
        at java.util.Properties$LineReader.readLine(Properties.java:434)
        at java.util.Properties.load0(Properties.java:353)                                                                                           
        at java.util.Properties.load(Properties.java:341)
        at io.radanalytics.examples.vertx.SparkPiVerticle.loadJarProperty(SparkPiVerticle.java:36)
        at io.radanalytics.examples.vertx.SparkPiVerticle.start(SparkPiVerticle.java:54)
        at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$163(DeploymentManager.java:429)
        at io.vertx.core.impl.ContextImpl.lambda$wrapTask$18(ContextImpl.java:335)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)                                                
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)                                        
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)                                                                              
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)                                              
        at java.lang.Thread.run(Thread.java:748)                                                                                                     
18/07/10 12:40:21 ERROR VertxIsolatedDeployer: Failed in deploying verticle                                                                          
java.lang.NullPointerException                                         
        at java.util.Properties$LineReader.readLine(Properties.java:434)                                                                             
        at java.util.Properties.load0(Properties.java:353)
        at java.util.Properties.load(Properties.java:341)       
        at io.radanalytics.examples.vertx.SparkPiVerticle.loadJarProperty(SparkPiVerticle.java:36)                                                   
        at io.radanalytics.examples.vertx.SparkPiVerticle.start(SparkPiVerticle.java:54)
        at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$163(DeploymentManager.java:429)
        at io.vertx.core.impl.ContextImpl.lambda$wrapTask$18(ContextImpl.java:335)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
        at java.lang.Thread.run(Thread.java:748)
java.lang.NullPointerException
        at java.util.Properties$LineReader.readLine(Properties.java:434)
        at java.util.Properties.load0(Properties.java:353)
        at java.util.Properties.load(Properties.java:341)
        at io.radanalytics.examples.vertx.SparkPiVerticle.loadJarProperty(SparkPiVerticle.java:36)
        at io.radanalytics.examples.vertx.SparkPiVerticle.start(SparkPiVerticle.java:54)
        at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$163(DeploymentManager.java:429)
        at io.vertx.core.impl.ContextImpl.lambda$wrapTask$18(ContextImpl.java:335)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
        at java.lang.Thread.run(Thread.java:748)
18/07/10 12:40:22 INFO ShutdownHookManager: Shutdown hook called
```

**Comment by ruivieira** [comment]
@elmiko thank you. ptal

File: assets/my-first-radanalytics-app/sparkpi-java-vertx.adoc [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+
+== Create the application resource files
+
+In addition to the source files we also need a few resource files to set default properties and configurations for our application. To begin creating your resource files you will first need to make a directory for them by running the following command from the root of your project:
+
+....
+mkdir -p src/main/resources
+....
+
+The first file you will create in that directory is named `application.properties` and it should contain the following contents:
```
Comment (elmiko) [inline assets/my-first-radanalytics-app/sparkpi-java-vertx.adoc]: in the reference repo this file is named `sparkpi.properties`, reverting the filename made my application run. i think this was the issue i ran into.

**Comment by ruivieira** [comment]
@elmiko well spotted! changed the filename.

**Comment by elmiko** [comment]
lgtm, thanks @ruivieira !

**Pull request merged**